src/qrexec-agent/qrexec-agent.c

```diff
@@ -926,9 +927,9 @@ static DWORD HandleDaemonHello(struct msg_header *header)
 
     if (info.version != QREXEC_PROTOCOL_VERSION)
     {
-        LogError("incompatible protocol version (%d instead of %d)",
+        LogWarning("incompatible protocol version (%d instead of %d)",
                  info.version, QREXEC_PROTOCOL_VERSION);
-        return ERROR_INVALID_FUNCTION;
+        //return ERROR_INVALID_FUNCTION;
```

Comment:
I'd prefer to change the condition to `info.version < QREXEC_PROTOCOL_VERSION` instead. The protocol got negotiation support, which works like this:
1. Both sides send their maximum supported version.
2. Both sides look at what other side sent them.
3. Choose minimum.
4. If that version is supported, continue, otherwise close the connection.

It means, the windows agent will be fine with sending version `2`, if the daemon sends at least `2` too.

When adding support for version `3`, it will need to either save what version was negotiated, or drop support for version `2`. But that's for another PR.